### ISSUE TITLE
[Bug] DGL Sparse Optimizer will crash when some DGL NodeEmbedding is not involved in the forward pass.

### ISSUE THREAD

**Issue by classicsong** [dmlc/dgl#2856]
## 🐛 Bug

The optimizer will crash if in a training iteration, one of the dgl NodeEmbedding is not touched (i.e., the embedding is not involved in farward), the optimizer will fail.

## To Reproduce

`
import torch as th

from dgl.nn import NodeEmbedding
from dgl.optim import SparseAdam, SparseAdagrad

num_embs = 10
emb_dim = 4
device=th.device('cpu')
dgl_emb = NodeEmbedding(num_embs, emb_dim, 'test')
th.nn.init.uniform_(dgl_emb.emb_tensor, 0, 1.0)
dgl_adam = SparseAdam(params=[dgl_emb], lr=0.01)

idx = th.randint(0, num_embs, size=(4,))
dgl_value = dgl_emb(idx, device).to(th.device('cpu'))
labels = th.ones((4,)).long()

dgl_adam.zero_grad()
dgl_loss = th.nn.functional.cross_entropy(dgl_value, labels)
dgl_loss.backward()

dgl_adam.step()
`

## Expected behavior

No fail

## Environment

 - DGL Version (e.g., 1.0): 0.6.1
 - Backend Library & Version (e.g., PyTorch 0.4.1, MXNet/Gluon 1.3): pytorch 1.6+
 - OS (e.g., Linux): Linux
 - How you installed DGL (`conda`, `pip`, source):
 - Build command you used (if compiling from source):
 - Python version:
 - CUDA/cuDNN version (if applicable):
 - GPU models and configuration (e.g. V100):
 - Any other relevant information:
